test/test_grammarconstraints.jl

```diff
@@ -0,0 +1,101 @@
+@testset verbose = true "AbstractGrammarConstraint" begin
+    @testset "utils" begin
+        @testset "_get_new_index" begin
+            rule = 8
+            mapping = Dict(1 => 10, 2 => 11, 3 => 12)
+            @test HerbConstraints._get_new_index(rule, mapping) == 8
+            rule = 3
+            @test HerbConstraints._get_new_index(rule, mapping) == 12
+        end
+    end
```

Comment:
Can remove these as well if you remove the function

Reply:
Done